pkg/oras/push_opts.go

```diff
@@ -53,6 +54,14 @@ func WithConfigAnnotations(annotations map[string]string) PushOpt {
 	}
 }
 
+// WithManifest overrides the manifest
```

Comment:
this doc should be a little more alarming that it'll invalidate other options like: opts.ManifestAnnotations etc...

Reply:
It should! Again copied from the `WithConfig` which does the same thing for the config manifest